packages/core/src/transaction.ts

```diff
@@ -193,12 +205,13 @@ export class Transaction {
   cleanup(): void {
     // Don't release the connection if there's a parent transaction or
     // if we've already cleaned up
-    if (this.parent || this.connection?.uuid === undefined) {
+    if (this.parent || this.#connection?.uuid === undefined) {
       return;
     }
 
-    this.sequelize.connectionManager.releaseConnection(this.connection);
-    this.connection.uuid = undefined;
+    this.sequelize.connectionManager.releaseConnection(this.#connection);
+    this.#connection.uuid = undefined;
```

Comment:
I don't think we need to set this to undefined if `this.#connection` itself is undefined in the following line

Reply:
Yes but the connection still exists, it's returned to the pool